### DISCUSSION TITLE
Exclude already assigned agents from the agent search on the user form

### DISCUSSION THREAD

**Discussion by tlammer** [specify/specify7#5590] [Feature Requests]
Currently, you can select in the query combobox, an agent that is already assigned to another user.  When you save the user, it asks you to select an agent again.  If you select the same agent again, then it says the agent is already used.
It would be smoother if already assigned users didn't appear at all.

**Comment by maxpatiiuk** [comment]
This would probably require a back-end endpoint for getting a list of all agents that are currently in use by someone.

Alternatively, this can be done by exposing to the front-end the table that stores that information

**Comment by benanhalt** [comment]
@maxpatiiuk Query combo boxes are currently getting converted into query-builder style queries behind the scenes right?  Why not just add a filter on specifyuser is null to the query?

That said, I'm not sure this is the best way to handle this. If the agent is already used by another user, but you don't know that, it could be confusing why an agent you know is in the database isn't appearing. Maybe the query cbx format could be made to include the name of the current specify user associated with the agent. It might even be possible by including agent.specifyuser.name in the query definition.
